### PR TITLE
[MAINTENANCE] Ensure that `DataContext.add_datasource` works with nested `DataConnector` ids

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Update existing tests to use nested `DataConnector` ids

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/data_context/conftest.py [hunk 1]
```diff
@@ -596,7 +611,7 @@ def _mocked_get_response(*args, **kwargs):
             {
                 "data": {
                     "attributes": {
-                        "datasource_config": datasource_config_with_names_and_ids,
+                        "datasource_config": datasource_config_with_names_and_ids.to_json_dict(),
```
Comment (cdkini) [inline tests/data_context/conftest.py:619]: This was a bug - the JSON response will not contain rich types

Comment (anthonyburdi) [reply tests/data_context/conftest.py:619]: Good catch!

**Review by anthonyburdi — APPROVED** [review]
LGTM!

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you for this @cdkini

File: tests/data_context/conftest.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply tests/data_context/conftest.py:619]: Yes, thank you!

**Pull request merged**